### PR TITLE
fixed dblp and scholar endpoints / fe integration with dblp and scholar endpoints

### PR DESCRIPTION
**Pull request by jsyska**

### CODE REVIEW

File: rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx [hunk 1]
```diff
@@ -108,25 +110,101 @@ const AuthorPopover: React.FC<{ author: Author }> = ({ author }) => {
...
+            });
+        }
+    };
+
+    const handleScholarClick = async () => {
+        if (scholarAuthor.data?.authors.length === 1) {
+            window.open(scholarAuthor.data.authors[0].scholar_url, '_blank');
+        } else {
+            console.log(scholarAuthor.data?.authors)
+            console.log(scholarAuthor.data?.authors.length)
```
Comment (CreativePhilip) [inline rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx:157]: leftover

File: rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx [hunk 2]
```diff
@@ -108,25 +110,101 @@ const AuthorPopover: React.FC<{ author: Author }> = ({ author }) => {
...
+                        {scholarAuthor.data?.authors.map(author => (
+                            <AuthorLink key={author.scholar_url}
+                                icon={<LinkOutlined />}
+                                label={authorName} onClick={() => window.open(author.scholar_url, '_blank')} />
+                        ))}
+                    </Flex>
+                ),
+                onOk() { },
+            });
+        }
```
Comment (CreativePhilip) [inline rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx:171]: This is practically exactly the same as it was for the dblp search, please extract it into a separate component

File: rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx [hunk 3]
```diff
@@ -108,25 +110,101 @@ const AuthorPopover: React.FC<{ author: Author }> = ({ author }) => {
...
+const useScholarAuthor = (author_name: string) => {
+    return useQuery({ queryKey: ["author_scholar", author_name], queryFn: () => getAuthorScholar(author_name), staleTime: 5 * 60 * 1000 })
+}
+
+const AuthorLinks: React.FC<{ author: Author }> = ({ author }) => {
+
+    const authorName = author.given_name + " " + author.surname;
+    const dblpAuthor = useDblpAuthor(authorName);
+    const scholarAuthor = useScholarAuthor(authorName);
+
```
Comment (CreativePhilip) [inline rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx:127]: To nicely enable preloading of these information, I would move the queries to the parent component, and just pass the whole object that you get from the useQuery

File: rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx [hunk 4]
```diff
@@ -108,25 +110,101 @@ const AuthorPopover: React.FC<{ author: Author }> = ({ author }) => {
...
+            {(resultsCount !== undefined) && <Typography.Text type="secondary">({resultsCount})</Typography.Text>}
         </Flex>
     );
+
 };
 
+
 const getInitials = (name: string | undefined) => {
     if (!name) return '';
     return name.split(' ').map(n => n[0]).join('');
```
Comment (CreativePhilip) [inline rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx:210]: This belongs to some soft og util file tbf

**Review by CreativePhilip — CHANGES_REQUESTED** [review]

File: rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx (hunk 3 above)
Comment (jsyska) [reply rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx:127]: Preloading information for all results might result in throttling due to the DBLP and Scholar API limitations. Considering the potential high number of authors in recommended articles, and that probably we won't need all of them, preloading may not be the best idea.

**Review by CreativePhilip — APPROVED** [review]

**Pull request merged**